test/actions/users.test.js

```diff
@@ -0,0 +1,107 @@
+
+import * as Actions from 'actions/users';
+import Client from 'client';
+import configureStore from 'store/configureStore';
+import {RequestStatus} from 'constants';
+import TestHelper from 'test_helper';
+
+describe('Actions.Users', () => {
+    it('login', (done) => {
+        TestHelper.initBasic(Client).then(async () => {
+            const store = configureStore();
+
+            store.subscribe(() => {
+                const loginRequest = store.getState().requests.users.login;
+                const currentUserId = store.getState().entities.users.currentId;
+                const profiles = store.getState().entities.users.profiles;
+                const preferences = store.getState().entities.users.myPreferences;
+
+                // TODO: uncomment when PLT-4167 is merged
+                // const teamMembers = store.getState().entities.teams.myMembers;
+
+                if (loginRequest.status === RequestStatus.SUCCESS || loginRequest.status === RequestStatus.FAILURE) {
+                    if (loginRequest.error) {
+                        done(new Error(loginRequest.error));
+                    } else {
+                        assert.ok(currentUserId);
+                        assert.ok(profiles);
+                        assert.ok(profiles[currentUserId]);
+                        assert.ok(preferences);
+
+                        // TODO: uncomment when PLT-4167 is merged
+                        // Object.keys(teamMembers).forEach((id) => {
+                        //     assert.ok(teamMembers[id].team_id);
+                        //     assert.equal(teamMembers[id].user_id, currentUserId);
+                        // });
+
+                        done();
+                    }
+                }
+            });
+
+            const user = TestHelper.basicUser;
+            await TestHelper.basicClient.logout();
+            Actions.login(user.email, 'password1')(store.dispatch, store.getState);
+        });
+    }).timeout(5000);
+
+    it('logout', (done) => {
+        TestHelper.initBasic(Client).then(async () => {
+            const store = configureStore();
+
+            store.subscribe(() => {
+                const logoutRequest = store.getState().requests.users.logout;
+                const general = store.getState().entities.general;
+                const users = store.getState().entities.users;
+                const loginView = store.getState().views.login;
+                const teams = store.getState().entities.teams;
+                const channels = store.getState().entities.channels;
+                const posts = store.getState().entities.posts;
+
+                if (logoutRequest.status === RequestStatus.SUCCESS || logoutRequest.status === RequestStatus.FAILURE) {
+                    if (logoutRequest.error) {
+                        done(new Error(logoutRequest.error));
+                    } else {
+                        assert.deepStrictEqual(general.config, {}, 'config not empty');
```

Comment:
right but when you logout is redirecting you to the selectServer page so you can switch servers...